rust/libnewsboat/src/filterparser.rs

```diff
@@ -173,7 +173,13 @@ pub fn parse(input: &str) -> Result<Expression, &str> {
         (parse)
     );
     match result {
-        Ok(expr) => Ok(expr.1),
+        Ok(expr) => {
+            if (expr.0.is_empty()) {
+                Ok(expr.1)
+            } else {
+                Err("Trailing characters")
```

Comment:
Sorry this didn't occur to me earlier, but: are these messages shown to the user? If so, let's use `gettextrs::gettext` here.

(Can't find these messages in `po/newsboat.pot`, so either they aren't shown, or weren't internationalized for the C++ version.)

Reply:
This isn't shown currently (still need to figure out how to return errors), and this error doesn't exist in our current code.
This is to avoid cases like `((a = b))` parsing and be consistent with our current parser.